### PR TITLE
Fixes for pybids dataset_description.json compatibility

### PR DESCRIPTION
**Pull request by akhanf**
Pybids is still a bit behind when it comes to parsing BIDS derivatives datasets, since the new BIDS spec (that e.g. fmriprep uses) uses a `GeneratedBy` list in the dataset_description.json, which supercedes the older `PIpelineDescription.Name` used by Pybids to parse a dataset. 

This PR adds a tool to add in a PipelineDescription to an existing dataset_description file, using the first entry of the GeneratedBy list to get the name and version. This is so we can more easily use BIDS derivatives in snakebids. Once Pybids is fixed to deal with this properly, we can ditch this, but it solves the problem for now. 

I've also added the PIpelineDescription field to our template project files, so that nakebids outputs are also compatible with pybids/snakebids when parsing as a derivative.

### CODE REVIEW

File: snakebids/fix_dd.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import json
+import argparse
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Update BIDS Derivatives Dataset  to be backwards compatible with pybids",
+        epilog=("Adds PipelineDescription.Name to the dataset_description.json file, from earlier BIDS spec) that pybids requires, using the first entry of the GeneratedBy list.  This is needed if parsing a modern version of fmriprep derivatives with snakebids/pybids."),
+    )
+    parser.add_argument("dataset_root", help="BIDS derivative dataset to fix")
```
Comment (pvandyken) [inline snakebids/fix_dd.py:12]: Not sure what the style has been in other parts of the app, but it's nice to make `dataset-root` an alias

**Review by tkkuehn — APPROVED** [review]
Looks good to me!

File: snakebids/fix_dd.py (hunk 1 above)
Comment (akhanf) [reply snakebids/fix_dd.py]: added

**Comment by akhanf** [comment]
merging this in, will have to address creating the dataset_description.json in another PR

**Pull request merged**